metals/src/main/scala/scala/meta/internal/metals/testProvider/TestDiscovery.scala

```diff
@@ -0,0 +1,45 @@
+package scala.meta.internal.metals.testProvider
+
+import javax.annotation.Nullable
+
+import scala.meta.internal.metals.MetalsEnrichments._
+
+import org.eclipse.{lsp4j => l}
+
+final case class TestDiscovery(
```

Comment:
Nit

```suggestion
final case class TestDiscoveryResult(
```

wdyt? No strong feeling on my part, I think it makes it a bit clearer though.

Reply:
Maybe even `TestSuitesDiscoveryResult` since it discovers whole suites?

Reply:
Maybe sure, I think so it's more clear that this is the result I think it'd be helpful. No strong feelings on the exact name.